src/Files.App/UserControls/NavigationToolbar.xaml.cs

```diff
@@ -258,32 +258,38 @@ private void ClickablePath_GettingFocus(UIElement sender, GettingFocusEventArgs
 
 		private async void Omnibar_QuerySubmitted(Omnibar sender, OmnibarQuerySubmittedEventArgs args)
 		{
-			if (Omnibar.CurrentSelectedMode == OmnibarPathMode)
+			var mode = Omnibar.CurrentSelectedMode;
+
+			// Path mode
+			if (mode == OmnibarPathMode)
 			{
 				await ViewModel.HandleItemNavigationAsync(args.Text);
 				(MainPageViewModel.SelectedTabItem?.TabItemContent as Control)?.Focus(FocusState.Programmatic);
+				return;
 			}
-			else if (Omnibar.CurrentSelectedMode == OmnibarCommandPaletteMode)
+
+			// Command palette mode
+			if (mode == OmnibarCommandPaletteMode)
```

Comment:
What's the reason to move away from "else if"?

Reply:
You can still use "else if" with the comment. if you use "if" here, even the current mode is Path mode, `mode == Palette` and `mode == Search` are evaludated and concequently takes a few milisec.